Lua/tests/syntax_test_lua.lua

```diff
@@ -0,0 +1,491 @@
+
+    function foo.bar:baz (...) end
+--  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ meta.function
+--           ^^^ variable.other - entity
+--              ^ punctuation.accessor
+--               ^^^ variable.other - entity
+--                  ^ punctuation.accessor
+--                   ^^^ entity.name.function
+--                       ^^^^^ meta.group
+--                             ^^^ keyword.control.end
+
+    function foo
+        .bar () end
+--      ^^^^^^^^^^^ meta.function
+--      ^ punctuation.accessor
+--       ^^^ entity.name.function
+
+    local function foo () end
+--  ^^^^^ storage.modifier
+--        ^^^^^^^^^^^^^^^^^^^ meta.function
+--                 ^^^ entity.name.function
+
+    ~function( a, b, ... ) end;
+--   ^^^^^^^^^^^^^^^^^^^^^^^^^ meta.function
+--   ^^^^^^^^ storage.type
+--           ^^^^^^^^^^^^^ meta.group
+--                         ^^^ keyword.control.end
+
+    ~function foo () end;
+--   ^^^^^^^^^^^^^^^^^^^ meta.function
+--            ^^^ entity.name.function
+
+    foo = function() end;
+--  ^^^ entity.name.function
+
+    foo.bar = function() end;
+--      ^^^ entity.name.function
+
+--STATEMENTS
+
+    end;
+--  ^^^ invalid.illegal.unexpected-end keyword.control.end
+
+    do
+--  ^^^ meta.block
+--  ^^ keyword.control
+        2 + 2
+    end
+-- ^^^^ meta.block
+--  ^^^ keyword.control.end
+
+    do 2 + 2 end
+--  ^^^^^^^^^^^^ meta.block
+--  ^^ keyword.control
+--           ^^^ keyword.control.end
+
+    if true then
+--  ^^ keyword.control.conditional
+--          ^^^^ keyword.control.conditional
+        2 + 2
+    elseif false then
+--  ^^^^^^ keyword.control.conditional
+--               ^^^^ keyword.control.conditional
+    else
+--  ^^^^ keyword.control.conditional
+    end
+
+
+    while true do
+--  ^^^^^ keyword.control.loop
+--             ^^ keyword.control
+        2 + 2
+    end
+--  ^^^ keyword.control.end
+
+    repeat
+--  ^^^^^^ keyword.control.loop
+        2 + 2;
+
+        end;
+--      ^^^ invalid.illegal.unexpected-end keyword.control.end
+    until true;
+--  ^^^^^ keyword.control.loop
+
+    for x = 1, y, z do end
+--  ^^^ keyword.control.loop
+--      ^ variable.other
+--        ^ keyword.operator.assignment
+--           ^ punctuation.separator.comma
+--             ^ variable.other
+--              ^ punctuation.separator.comma
+--                ^ variable.other
+--                  ^^^^^^ meta.block
+--                  ^^ keyword.control
+--                     ^^^ keyword.control.end
+
+    for x, y in a, b do end
+--  ^^^ keyword.control.loop
+--      ^ variable.other
+--       ^ punctuation.separator.comma
+--         ^ variable.other
+--           ^^ keyword.control.loop
+--              ^ variable.other
+--               ^ punctuation.separator.comma
+--                 ^ variable.other
+--                   ^^^^^^ meta.block
+--                   ^^ keyword.control
+--                      ^^^ meta.block keyword.control.end
```

Comment:
(nitpick) `meta.block` is tested twice here.

Reply:
Fixed.